exonum/src/api/websocket.rs

```diff
@@ -33,20 +137,33 @@ pub(crate) struct Message(pub String);
 #[rtype(usize)]
 pub(crate) struct Subscribe {
     pub address: Recipient<Message>,
+    pub sub_types: Vec<SubscriptionType>,
 }
 
 #[derive(Message)]
 pub(crate) struct Unsubscribe {
     pub id: usize,
 }
 
+#[derive(Message)]
+pub(crate) struct UpdateSubscriptions {
+    pub id: usize,
+    pub sub_types: Vec<SubscriptionType>,
+}
+
 #[derive(Message)]
 pub(crate) struct Broadcast {
     pub block_hash: Hash,
 }
 
+#[derive(Message)]
+#[rtype("Result<TransactionResponse, failure::Error>")]
+pub(crate) struct Transaction {
+    tx: TransactionHex,
+}
+
 pub(crate) struct Server {
-    pub subscribers: HashMap<usize, Recipient<Message>>,
+    pub subscribers: HashMap<SubscriptionType, HashMap<usize, Recipient<Message>>>,
```

Comment:
Subscription types are not many, and `HashMap` is less effective for them than the simple `BTreeMap`.

Reply:
Done